components/server/src/model/transformations.py

```diff
@@ -18,6 +18,21 @@ def hide_credentials(data_model, *reports) -> None:
                 source["parameters"][parameter_key] = "this string replaces credentials"
 
 
+def encrypt_credentials(data_model, public_key: str, *reports: dict):
+    """Encrypt all credentials in the reports."""
+    for source in iter_sources(reports):
+        for parameter_key, parameter_value in source.get("parameters", {}).items():
+            if parameter_value and is_password_parameter(data_model, source["type"], parameter_key):
+                password = source["parameters"][parameter_key]
+                if isinstance(password, (dict, list)):
+                    password = json.dumps(password)
+
+                password_bytes = password.encode()
+                key_encrypted_value = asymmetric_encrypt(public_key.encode(), password_bytes)
```

Comment:
Why is this variable called "key_encrypted_value" and not simply "encrypted_value"?

Reply:
Because it's only the key of the symmetric encryption that is encrypted, not the value of the message itself. 
But indeed, maybe something like encrypted_fernet_key would be more descriptive.

Reply:
disregard my previous comment, it's called like this because this function returns a tuple with the rsa encrypted fernet key and the fernet (symmetric) encrypted value of the password.